server/controllers/article.js

```diff
@@ -59,18 +56,47 @@ class Article {
    * @returns {object} User Object
    */
   static async getArticles(request, response) {
-    try {
-      const limit = parseInt(request.query.limit, 10) || 10;
-      const page = parseInt(request.query.page, 10) || 1;
+    const limit = parseInt(request.query.limit, 10) || 10;
+    const page = parseInt(request.query.page, 10) || 1;
 
-      const articles = await articleRepo.getArticles(limit, page);
-      if (!articles) {
-        return goodHttpResponse(response, 200, 'no articles found');
-      }
-      return paginatedHttpResponse(response, 200, 'all articles', articles);
-    } catch (error) {
-      return badHttpResponse(response, 500, 'There was an internal server error', error);
+    const articles = await articleRepo.getArticles(limit, page);
+    if (!articles) {
+      return goodHttpResponse(response, 200, 'no articles found');
     }
+    return paginatedHttpResponse(response, 200, 'all articles', articles);
+  }
+
+  /**
+   * Post a complaint on an article
+   * @param {object} request Request Object
+   * @param {object} response Response Object
+   * @returns {object} Error object if request was not successful
+   * or posted complaint object if successful
+   */
+  static async postComplaint(request, response) {
+    const { userId, article } = request;
+    const { complaintBody, complaintType } = request.body;
+
+    if (article.userid === userId) {
+      return badHttpResponse(
+        response,
+        400,
+        'What is wrong with you? Why do you want to raise a complaint against yourself',
```

Comment:
This error message is too harsh lol. You can try this "Please reconsider raising a complaint against yourself".

Reply:
Lol. I wasn't supposed to leave this here. Thanks. I will address it.